source/ed/osm2ed.h

```diff
@@ -372,18 +372,34 @@ struct Rect{
 
 enum class OsmObjectType {
     Node,
-    Way
+    Way,
+    Relation,
 };
 
 inline std::string to_string(OsmObjectType t) {
     switch (t) {
-    case OsmObjectType::Node: return "n";
-    case OsmObjectType::Way: return "w";
+    case OsmObjectType::Node : return "node";
+    case OsmObjectType::Way : return "way";
+    case OsmObjectType::Relation : return "relation";
     default:
         throw navitia::exception("not implemented");
     }
 }
 
+class OsmPoi : public ed::types::Poi
```

Comment:
if i understand it right, you use the class only for constructing the id... maybe a free function can do the job and it'll be simpler

Reply:
Maybe then use the constructor from parent class?
Although, the typing can help not messing OSM and custom POI use/creation...

Reply:
@xlqian - I started with a free function, but we have 2 different behaviors with different parameter types for `Poi` and `OsmPoi`. The problem is that we also have some common logic between the 2 objects (both `uri`s start with `"poi:"`, hence the class design.
@pbougue - I do use the parent constructor. When you create an `OsmPoi`, `Poi::uri` is being set through the default constructor (eg `Poi::Poi() = default;`).